services/audit-service/src/main/java/datawave/microservice/audit/replay/config/ReplayConfig.java

```diff
@@ -0,0 +1,39 @@
+package datawave.microservice.audit.replay.config;
+
+import datawave.microservice.audit.replay.status.StatusCache;
+import datawave.microservice.audit.replay.util.ConcurrentMapCacheInspector;
+import datawave.microservice.cached.CacheInspector;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.cache.concurrent.ConcurrentMapCacheManager;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+
+@Configuration
+@EnableCaching
+@EnableConfigurationProperties(ReplayProperties.class)
+@ConditionalOnProperty(name = "audit.replay.enabled", havingValue = "true")
+public class ReplayConfig {
+    
+    @Bean
+    public ThreadPoolTaskExecutor auditReplayExecutor(ReplayProperties replayProperties) {
```

Comment:
If the context is refreshed, this may cause thread leaks when this bean is re-created. May need to look into that.

Reply:
I'm working on a test for the next commit.

Reply:
I tested this.  As it is now, nothing happens on a refresh event.  However, if I mark the ThreadPoolTaskExecutor as @RefreshScope, any running replays are killed.  That's ok though, because eventually they will be marked as IDLE, at which point they can be resumed.  

For now, I am going to leave the RefreshScope annotation out.  What I will do instead is create a separate ticket to go over the entire audit service and try to get it configured to handle refresh appropriately.